src/library.js

```diff
@@ -3118,13 +3118,27 @@ LibraryManager.library = {
     buf >>= 2;
     while (ch = HEAPU8[sigPtr++]) {
 #if ASSERTIONS
-      assert(ch === 100/*'d'*/ || ch === 102/*'f'*/ || ch === 105 /*'i'*/);
+      assert(ch === 100/*'d'*/ || ch === 102/*'f'*/ || ch === 105 /*'i'*/ || ch === 106 /*'j'*/);
+#if !WASM_BIGINT
+      assert(ch !== 106/*'j'*/, "i64 arguments to ASM_JS function are not available without WASM_BIGINT");
+#endif
 #endif
       // A double takes two 32-bit slots, and must also be aligned - the backend
-      // will emit padding to avoid that.
-      var readAsmConstArgsDouble = ch < 105;
+      // will emit padding to avoid that.  Same with i64.
+      var readAsmConstArgsDouble = ch != 105
+      var v;
       if (readAsmConstArgsDouble && (buf & 1)) buf++;
-      readAsmConstArgsArray.push(readAsmConstArgsDouble ? HEAPF64[buf++ >> 1] : HEAP32[buf]);
+#if WASM_BIGINT
+      if (ch === 106) {
+        v = HEAP64[buf++ >> 1];
+      } else
+#endif
+      if (readAsmConstArgsDouble) {
```

Comment:
```suggestion
      if (ch == 105) {
```
This seems clearer as it avoids ambiguity around i64, I think.